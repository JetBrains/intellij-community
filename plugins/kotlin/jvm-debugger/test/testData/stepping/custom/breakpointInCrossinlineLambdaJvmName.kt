package breakpointInCrossinlineLambdaJvmName

fun main() {
    // STEP_INTO: 1
    // RESUME: 1
    //Breakpoint!, lambdaOrdinal = 1
    foo { f() }

    // STEP_INTO: 1
    // RESUME: 1
    //Breakpoint!, lambdaOrdinal = 2
    foo { f() }; foo { g() }
}

@JvmName("boo")
inline fun foo(crossinline body: () -> Unit) {
    val runnable = object : Runnable {
        override fun run() = body()
    }
    runnable.run()
}

fun f() = Unit
fun g() = Unit
