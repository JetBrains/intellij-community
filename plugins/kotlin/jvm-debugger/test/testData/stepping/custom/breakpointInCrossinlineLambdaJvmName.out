LineBreakpoint created at breakpointInCrossinlineLambdaJvmName.kt:7 lambdaOrdinal = 1
LineBreakpoint created at breakpointInCrossinlineLambdaJvmName.kt:12 lambdaOrdinal = 2
Run Java
Connected to the target VM
breakpointInCrossinlineLambdaJvmName.kt:7
breakpointInCrossinlineLambdaJvmName.kt:23
breakpointInCrossinlineLambdaJvmName.kt:12
breakpointInCrossinlineLambdaJvmName.kt:24
Disconnected from the target VM

Process finished with exit code 0
